dts/arm/st/h7rs/stm32h7rs.dtsi

```diff
@@ -0,0 +1,831 @@
+			interrupt-names = "line0", "line1", "line2", "line3",
+					  "line4", "line5", "line6", "line7",
+					  "line8", "line9", "line10", "line11",
+					  "line12", "line13", "line14", "line15";
+			line-ranges = <0 1>, <1 1>, <2 1>, <3 1>,
+				      <4 1>, <5 1>, <6 1>, <7 1>,
+				      <8 1>, <9 1>, <10 1>, <11 1>,
+				      <12 1>, <13 1>, <14 1>, <15 1>;
+		};
+
+		pinctrl: pin-controller@58020000 {
+			compatible = "st,stm32-pinctrl";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0x58020000 0x2400>;
+
+			gpioa: gpio@58020000 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58020000 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00000001>;
+			};
+
+			gpiob: gpio@58020400 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58020400 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00000002>;
+			};
+
+			gpioc: gpio@58020800 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58020800 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00000004>;
+			};
+
+			gpiod: gpio@58020C00 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58020C00 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00000008>;
+			};
+
+			gpioe: gpio@58021000 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58021000 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00000010>;
+			};
+
+			gpiof: gpio@58021400 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58021400 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00000020>;
+			};
+
+			gpiog: gpio@58021800 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58021800 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00000040>;
+			};
+
+			gpioh: gpio@58021c00 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58021c00 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00000080>;
+			};
+
+			gpiom: gpio@58023000 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58023000 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00001000>;
+			};
+
+			gpion: gpio@58023400 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58023400 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00002000>;
+			};
+
+			gpioo: gpio@58023800 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58023800 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00004000>;
+			};
+
+			gpiop: gpio@58023c00 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x58023c00 0x400>;
+				clocks = <&rcc STM32_CLOCK_BUS_AHB4 0x00008000>;
+			};
+		};
+
+		iwdg: iwdg1: watchdog@58004800 {
```

Comment:
Please limit dtsi file to basic nodes for now.
Then we'll add new hw block nodes by sequence with matching board configuration and test when possible

Reply:
removing adc, wdg, rng, dma nodes from this first PR